### ISSUE TITLE
SVG issue with 1.2.0

### ISSUE THREAD

**Issue by dandar3** [gcacace/android-signaturepad#59]
I just noticed an issue with the SVG generated by 1.2.0 that I could not replicate with 1.1.0 - it seems to connect on its own to the top left corner, see the two SVG / emulator screenshots with the different versions. The signature is a simple series of loops starting from somewhere in the top left corner.

(PS: It appears there is a difference in default line thinkness between the two versions, I can raise that as a different issue, I left it with defaults just in case it might be relevant somehow).

Comparing https://github.com/gcacace/android-signaturepad/compare/v1.1.0...v1.2.0 I see there are quite a few SVG related changes mostly done by @thomasleveil, copying him for thoughts on this.

Tested with Visual Studio Emulator for Android
4.5", xhdpi, Android 5.1.1 (similar to Motorola Moto G, Motorola Moto X)

[screenshots + svgs.zip](https://github.com/gcacace/android-signaturepad/files/339912/screenshots.svgs.zip)

**Comment by thomasleveil** [comment]
@dandar3 I was unable to reproduce this issue using [the example app](https://github.com/gcacace/android-signaturepad/tree/e09688542f11777baa550018cff9c3de2d659252/SignaturePad-Example) @f890668 running on a physical device (Nexus 5 / Android 6.0.1).

Can you try to reproduce with when :
- using the example app
- using a physical device

**Comment by dandar3** [comment]
@thomasleveil Thanks for having a look at this.

I was able to reproduce it on a physical device (Nexus 7 / Android 5.1.1), but indeed it doesn't seem to happen every time, so maybe try a few times.

I made some small adjstments to the sample app to use the release version instead of the library source code to avoid any intererence and added some code to the "Save signature" button to show the SVG afterwards. 

```
 dependencies {
-    compile project(":signature-pad")
+    compile 'com.github.gcacace:signature-pad:1.2.0'
}
```
```
        mSaveButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                 [...]
+                try {
+                    Log.e("SVG", mSignaturePad.getSignatureSvg());
+
+                    File        file   = new File("/sdcard/signature.svg");
+                    PrintStream stream = new PrintStream(new FileOutputStream(file));
+                    stream.print(mSignaturePad.getSignatureSvg());
+                    stream.close();
+
+                    Intent intent = new Intent(Intent.ACTION_VIEW);
+                    intent.setDataAndType(Uri.fromFile(file), "image/svg+xml");
+                    startActivity(intent);
+                }
+                catch (Exception e) {
+                    Log.e(getClass().getSimpleName(), Log.getStackTraceString(e));
+                }
            }
```

Attaching a screen recording to show the problem.
[SVG issue.zip](https://github.com/gcacace/android-signaturepad/files/341225/SVG.issue.zip)

**Comment by mariano-huerta** [comment]
I can confirm the issue in a physical device (One plus one / Android 6.0.1). It seems to happen when you tap more than once on the pad.

**Comment by dnimon** [comment]
I can also confirm this issue on both Android 6.0 and Android 4.4.4. PNG works fine.

**Comment by MrXavierLin** [comment]
Same here, 1.2.0 saving to SVG with unwanted lines from up-left corner, Android 5.0.1 simulator

**Comment by rclements** [comment]
I'm also having this problem on Android 6.0, any idea what is happening here?

**Comment by adamcoulter** [comment]
What I seem to be getting are some paths that are missing the "c" curve command and because the points are relative the path gets rendered up to the top left corner. It looks like an issue in the SVGPathBuilder.

**Comment by crm76** [comment]
I have also experienced this issue. Definitely a cool library otherwise; thank you for your shared efforts. Unfortunately this bug makes it unusable for my purposes.

Generated SVG and corresponding render:

`<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.2" baseProfile="tiny" height="488" width="1240">
    <g stroke-linejoin="round" stroke-linecap="round" fill="none" stroke="black">
        <path stroke-width="9" d="M227 118" />
        <path stroke-width="8" d="M227 118c0 8-1 8 0 17" />
        <path stroke-width="6" d="M227 135c2 20 4 19 5 39 4 40 3 40 5 80 1 19 0 19 0 38 0 31 1 31 0 62-1 14-2 14-5 27" />
        <path stroke-width="7" d="M232 381c0 2-1 2-1 4" />
        <path stroke-width="8" d="M115 141 117-25 117-25 234-50" />
        <path stroke-width="6" d="M419 268 4 0 4 1 8 0 6-2 6-2 12-4 6-2 6-1 12-4 8-2 8-2 16-6 8-3 8-3 16-6 8-4 8-4 16-9 7-4 7-4 13-10 6-5 6-4 10-10 3-4 4-4 5-9 0-4 0-6-3-9-3-5-4-5-10-8-7-5-7-5-14-7-9-4-9-5-19-6-10-2-10-1-20 0-10 0-11 0-21 2-10 3-10 4-19 9-10 4-10 4-19 10-8 6-9 7-16 15-7 7-7 8-12 16-6 9-6 9-9 19-3 10-2 10-3 20-1 10-3 11 0 21 2 10 4 9 10 19 5 9 5 9 11 17 7 7 6 8 14 14 8 6 8 7 17 11 9 3 10 3 19 4 10 1 10 1 21-1 10-1 10-2 20-4 9-2 9-3 18-5 9-2 9-2 18-4" />
        <path stroke-width="6" d="M711 155-4-2-4-3-9-4-2-1-2-1-4 0-6 0-6 0-11 2-8 2-8 2-15 6-9 5-8 6-17 13-9 7-9 7-19 15-9 8-10 8-18 17-7 7-7 8-12 16-5 6-5 5-9 12-2 6-4 7-3 12 0 4 2 4 6 7 6 5 6 5 13 8 9 3 9 3 18 5 9 2 10 1 19 4 10 3 10 3 19 7 8 3 9 3 16 7 7 3 9 2 12 8 3 4 3 6 1 11-3 8-4 8-11 16-7 9-8 8-17 17-12 11-12 12-24 22-15 12-16 11-31 22-14 11-15 10-29 21-9 7-8 8-17 16-4 3-4 3-8 6" />
        <path stroke-width="6" d="M848 69 0 4 0 4 0 8 0 5 0 5 0 9 0 10 1 10 0 20-1 17-2 17-2 34-2 32-1 32-2 63 0 39-1 39 0 78 1 25 3 25 6 50 2 16 2 16 4 32 2 13 1 13 4 25 2 8 2 8 4 15 1 4 1 4 2 8" />
        <path stroke-width="6" d="M752 139 6 3 6 4 12 6 7 2 7 1 14 1 11 1 11 1 22 1 14 0 14 1 28 1 21-1 21-2 42-4 21-2 21-2 43-3 18-1 18 0 37 0" />
    </g>
</svg>`

![image](https://cloud.githubusercontent.com/assets/10379425/21546075/a77bf24a-cd99-11e6-8268-ca7ed0c9ada5.png)

Manually corrected (added the missing "c" curve command) SVG and corresponding render:

`<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.2" baseProfile="tiny" height="488" width="1240">
    <g stroke-linejoin="round" stroke-linecap="round" fill="none" stroke="black">
        <path stroke-width="9" d="M227 118c" />
        <path stroke-width="8" d="M227 118c0 8-1 8 0 17" />
        <path stroke-width="6" d="M227 135c2 20 4 19 5 39 4 40 3 40 5 80 1 19 0 19 0 38 0 31 1 31 0 62-1 14-2 14-5 27" />
        <path stroke-width="7" d="M232 381c0 2-1 2-1 4" />
        <path stroke-width="8" d="M115 141c 117-25 117-25 234-50" />
        <path stroke-width="6" d="M419 268c 4 0 4 1 8 0 6-2 6-2 12-4 6-2 6-1 12-4 8-2 8-2 16-6 8-3 8-3 16-6 8-4 8-4 16-9 7-4 7-4 13-10 6-5 6-4 10-10 3-4 4-4 5-9 0-4 0-6-3-9-3-5-4-5-10-8-7-5-7-5-14-7-9-4-9-5-19-6-10-2-10-1-20 0-10 0-11 0-21 2-10 3-10 4-19 9-10 4-10 4-19 10-8 6-9 7-16 15-7 7-7 8-12 16-6 9-6 9-9 19-3 10-2 10-3 20-1 10-3 11 0 21 2 10 4 9 10 19 5 9 5 9 11 17 7 7 6 8 14 14 8 6 8 7 17 11 9 3 10 3 19 4 10 1 10 1 21-1 10-1 10-2 20-4 9-2 9-3 18-5 9-2 9-2 18-4" />
        <path stroke-width="6" d="M711 155c-4-2-4-3-9-4-2-1-2-1-4 0-6 0-6 0-11 2-8 2-8 2-15 6-9 5-8 6-17 13-9 7-9 7-19 15-9 8-10 8-18 17-7 7-7 8-12 16-5 6-5 5-9 12-2 6-4 7-3 12 0 4 2 4 6 7 6 5 6 5 13 8 9 3 9 3 18 5 9 2 10 1 19 4 10 3 10 3 19 7 8 3 9 3 16 7 7 3 9 2 12 8 3 4 3 6 1 11-3 8-4 8-11 16-7 9-8 8-17 17-12 11-12 12-24 22-15 12-16 11-31 22-14 11-15 10-29 21-9 7-8 8-17 16-4 3-4 3-8 6" />
        <path stroke-width="6" d="M848 69c 0 4 0 4 0 8 0 5 0 5 0 9 0 10 1 10 0 20-1 17-2 17-2 34-2 32-1 32-2 63 0 39-1 39 0 78 1 25 3 25 6 50 2 16 2 16 4 32 2 13 1 13 4 25 2 8 2 8 4 15 1 4 1 4 2 8" />
        <path stroke-width="6" d="M752 139c 6 3 6 4 12 6 7 2 7 1 14 1 11 1 11 1 22 1 14 0 14 1 28 1 21-1 21-2 42-4 21-2 21-2 43-3 18-1 18 0 37 0" />
    </g>
</svg>`

![image](https://cloud.githubusercontent.com/assets/10379425/21546140/0184df9a-cd9a-11e6-9a64-648bec268f4f.png)

**Comment by adamcoulter** [comment]
I have fixed this and it is ready to review & merge https://github.com/gcacace/android-signaturepad/pull/71

**Comment by gcacace** [comment]
Bugfix addressed in release 1.2.1
